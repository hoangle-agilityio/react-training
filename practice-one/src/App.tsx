import React from "react";
import TopNav from "./components/TopNav";

export default function App(): JSX.Element {
  return (
    <TopNav />
  );
}
