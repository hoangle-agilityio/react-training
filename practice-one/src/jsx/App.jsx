import React from "react";
import TopNav from "./components/TopNav";

export default function App() {
  return (
    <TopNav />
  );
}
