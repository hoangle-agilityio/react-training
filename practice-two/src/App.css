.app {
  background-color: #eeeeee;
  padding: 30px 0;
}

.app__heading {
  font-size: 36px;
  text-align: center;
  text-transform: uppercase;
  color: #333333;
  margin-bottom: 30px;
}

.user-list {
  border-collapse: collapse;
  width: 90%;
  margin: 0 auto;
}

.list-head,
.list-item {
  border: 1px solid #333333;
  text-align: left;
  padding: 8px;
}

.empty-item,
.error-msg {
  color: #ff0000;
  font-style: italic;
}

.create-user {
  width: 90%;
  margin: 0 auto;
  text-align: right;
}

.search-user {
  width: 90%;
  margin: 0 auto;
  text-align: right;
  margin-bottom: 30px;
}

.search-user input {
  padding: 5px;
  margin-left: 10px;
}

.loader {
  border: 16px solid #f3f3f3;
  border-radius: 50%;
  border-top: 16px solid #3498db;
  width: 120px;
  height: 120px;
  -webkit-animation: spin 2s linear infinite; /* Safari */
  animation: spin 2s linear infinite;
  position: absolute;
  top: 25%;
  left: 50%;
}

/* Safari */
@-webkit-keyframes spin {
  0% {
    -webkit-transform: rotate(0deg);
  }
  100% {
    -webkit-transform: rotate(360deg);
  }
}

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}
