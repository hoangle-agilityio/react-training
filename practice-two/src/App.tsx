import { useEffect, useState } from "react";
import "./App.css";
import Button from "./components/Button";
import ModalUser from "./components/Modal";
import { searching } from "./core/helpers/search-helper";
import User from "./core/interfaces/user";
import { deleteUser, getAllUsers } from "./services";

export default function App(): JSX.Element {
  const [users, setUsers] = useState<User[]>([]);
  const [currentUser, setCurrentUser] = useState<User>();
  const [isOpenModal, setIsOpenModal] = useState(false);
  const [searchTerm, setSearchTerm] = useState("");
  const [isViewUser, setIsViewUser] = useState(false);

  const handleOpenModal = (): void => setIsOpenModal(true);
  const handleCloseModal = (): void => setIsOpenModal(false);

  const openModalUser = (user?: User, isView: boolean = false) => {
    setIsViewUser(isView);
    setCurrentUser(user);
    handleOpenModal();
  }

  // Get all users
  const handleGetUsers = async (): Promise<void> => {
    try {
      setUsers(await getAllUsers());
    } catch (error) {
      throw new Error(`Get data failed: ${error}`);
    }
  };

  // Delete user when click to button delete
  const handleDeleteUser = async (user: User): Promise<void> => {
    if (confirm("Are you sure to delete this user?")) {
      try {
        await deleteUser(user.id);

        // Find index of user
        const index = users.indexOf(user);

        // If index exists, remove that index in array.
        // After that, set state user list after remove index
        if (index > -1) {
          users.splice(index, 1);
          setUsers([...users]);
        }

        alert("User deleted successfully!");
      } catch (error) {
        throw new Error(`Delete data failed: ${error}`);
      }
    }
  }

  // Filter user list by searchTerm
  const searchResult = users.filter(user => {
    return searching(user, searchTerm);
  });

  // Use useEffect avoid to call function repeatedly
  useEffect(() => {
    handleGetUsers();
  }, []);

  return (
    <div className="app">
      <h1 className="app__heading">User Management</h1>
      <section className="create-user">
        <Button
          buttonName="Add User"
          type="success"
          onClick={() => openModalUser(undefined)}
        />
      </section>
      <section className="search-user">
        <label htmlFor="search">Search:</label>
        <input type="text" id="search" onChange={event => setSearchTerm(event.target.value)} />
      </section>
      <section>
        <table className="user-list">
          <thead>
            <tr>
              <th className="list-head">User Id</th>
              <th className="list-head">Name</th>
              <th className="list-head">Email</th>
              <th className="list-head">Action</th>
            </tr>
          </thead>
          <tbody>
            {searchResult.length === 0 ? (
              <tr>
                <td className="empty-item">No data found!</td>
              </tr>
            ) : (
              <>
                {searchResult.map(user => (
                  <tr key={user.id}>
                    <td className="list-item">{user.id}</td>
                    <td className="list-item">{user.name}</td>
                    <td className="list-item">{user.email}</td>
                    <td className="list-item">
                      <Button
                        buttonName="View"
                        type="info"
                        onClick={() => openModalUser(user, true)}
                      />

                      <Button
                        buttonName="Edit"
                        type="primary"
                        onClick={() => openModalUser(user)}
                      />

                      <Button
                        buttonName="Delete"
                        type="danger"
                        onClick={() => handleDeleteUser(user)}
                      />
                    </td>
                  </tr>
                ))}
              </>
            )}
          </tbody>
        </table>
      </section>
      <ModalUser
        open={isOpenModal}
        currentUser={currentUser}
        users={users}
        isViewUser={isViewUser}
        onEdit={setIsViewUser}
        onSuccess={setUsers}
        closeModal={handleCloseModal}
      />
    </div>
  );
}
