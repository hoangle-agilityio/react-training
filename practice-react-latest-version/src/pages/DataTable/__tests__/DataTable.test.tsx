import * as React from "react";
import App from "..";
import { act, fireEvent, render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "react-query";

// Create mock for api
jest.mock("services/customers");

const queryClient = new QueryClient();

describe("Data Table Page", () => {
  test("Should match snapshot for Data Table", async () => {
    const dom = await act(() => {
      return render(
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      );
    });
    expect(dom).toMatchSnapshot();
  });

  test("Should search with the value entered", async () => {
    const dom = await act(() => {
      return render(
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      );
    });
    const input = dom.getByPlaceholderText("Search...") as HTMLInputElement;
    const value = "description 11";

    fireEvent.change(input, { target: { value } });

    expect(input.value).toBe(value);
    expect(dom).toMatchSnapshot();
  });

  test("Should sort by name", async () => {
    const dom = await act(() => {
      return render(
        <QueryClientProvider client={queryClient}>
          <App />
        </QueryClientProvider>
      );
    });

    const name = dom.getByText("Name");

    fireEvent.click(name);
    expect(dom).toMatchSnapshot();
  });
});
