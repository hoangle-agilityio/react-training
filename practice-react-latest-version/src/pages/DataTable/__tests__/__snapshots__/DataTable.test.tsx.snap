// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Table Page Should match snapshot for Data Table 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-f7fu58"
      >
        <div
          class="css-8zasf"
        >
          <div
            class="css-1wo2j8q"
          >
            <div
              class="css-1anpw3e"
            >
              <svg
                class="chakra-icon css-tud5pv"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
              <input
                class="chakra-input css-oce1e7"
                id="search:r0:"
                placeholder="Search..."
                value=""
              />
            </div>
          </div>
          <button
            class="chakra-button css-11ko6ne"
            type="button"
          >
            + Add customer
          </button>
        </div>
        <div
          class="css-1milhtq"
        >
          <div
            class="css-1dj3a23"
          >
            <p
              class="chakra-text css-0"
            >
              Name
            </p>
            <img
              class="chakra-image css-0"
              src="/icons/column-sorting-none.svg"
            />
          </div>
          <p
            class="chakra-text css-1i8mt9"
          >
            Description
          </p>
          <p
            class="chakra-text css-l7sbe5"
          >
            Status
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Rate
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Balance
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Deposit
          </p>
          <div
            class="css-yikrym"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-f7fu58"
    >
      <div
        class="css-8zasf"
      >
        <div
          class="css-1wo2j8q"
        >
          <div
            class="css-1anpw3e"
          >
            <svg
              class="chakra-icon css-tud5pv"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
            <input
              class="chakra-input css-oce1e7"
              id="search:r0:"
              placeholder="Search..."
              value=""
            />
          </div>
        </div>
        <button
          class="chakra-button css-11ko6ne"
          type="button"
        >
          + Add customer
        </button>
      </div>
      <div
        class="css-1milhtq"
      >
        <div
          class="css-1dj3a23"
        >
          <p
            class="chakra-text css-0"
          >
            Name
          </p>
          <img
            class="chakra-image css-0"
            src="/icons/column-sorting-none.svg"
          />
        </div>
        <p
          class="chakra-text css-1i8mt9"
        >
          Description
        </p>
        <p
          class="chakra-text css-l7sbe5"
        >
          Status
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Rate
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Balance
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Deposit
        </p>
        <div
          class="css-yikrym"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Data Table Page Should search with the value entered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-f7fu58"
      >
        <div
          class="css-8zasf"
        >
          <div
            class="css-1wo2j8q"
          >
            <div
              class="css-1anpw3e"
            >
              <svg
                class="chakra-icon css-tud5pv"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
              <input
                class="chakra-input css-oce1e7"
                id="search:r1:"
                placeholder="Search..."
                value="description 11"
              />
            </div>
          </div>
          <button
            class="chakra-button css-11ko6ne"
            type="button"
          >
            + Add customer
          </button>
        </div>
        <div
          class="css-1milhtq"
        >
          <div
            class="css-1dj3a23"
          >
            <p
              class="chakra-text css-0"
            >
              Name
            </p>
            <img
              class="chakra-image css-0"
              src="/icons/column-sorting-none.svg"
            />
          </div>
          <p
            class="chakra-text css-1i8mt9"
          >
            Description
          </p>
          <p
            class="chakra-text css-l7sbe5"
          >
            Status
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Rate
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Balance
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Deposit
          </p>
          <div
            class="css-yikrym"
          />
        </div>
        <div
          class="css-3ehqor"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            class="chakra-progress css-7d3cb9"
            data-indeterminate=""
            role="progressbar"
          >
            <svg
              class="css-tn9112"
              viewBox="0 0 100 100"
            >
              <circle
                class="chakra-progress__track css-1bhxzgy"
                cx="50"
                cy="50"
                r="42"
                stroke-width="10px"
              />
              <circle
                class="chakra-progress__indicator css-qa0lra"
                cx="50"
                cy="50"
                r="42"
                stroke-width="10px"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-f7fu58"
    >
      <div
        class="css-8zasf"
      >
        <div
          class="css-1wo2j8q"
        >
          <div
            class="css-1anpw3e"
          >
            <svg
              class="chakra-icon css-tud5pv"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
            <input
              class="chakra-input css-oce1e7"
              id="search:r1:"
              placeholder="Search..."
              value="description 11"
            />
          </div>
        </div>
        <button
          class="chakra-button css-11ko6ne"
          type="button"
        >
          + Add customer
        </button>
      </div>
      <div
        class="css-1milhtq"
      >
        <div
          class="css-1dj3a23"
        >
          <p
            class="chakra-text css-0"
          >
            Name
          </p>
          <img
            class="chakra-image css-0"
            src="/icons/column-sorting-none.svg"
          />
        </div>
        <p
          class="chakra-text css-1i8mt9"
        >
          Description
        </p>
        <p
          class="chakra-text css-l7sbe5"
        >
          Status
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Rate
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Balance
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Deposit
        </p>
        <div
          class="css-yikrym"
        />
      </div>
      <div
        class="css-3ehqor"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="chakra-progress css-7d3cb9"
          data-indeterminate=""
          role="progressbar"
        >
          <svg
            class="css-tn9112"
            viewBox="0 0 100 100"
          >
            <circle
              class="chakra-progress__track css-1bhxzgy"
              cx="50"
              cy="50"
              r="42"
              stroke-width="10px"
            />
            <circle
              class="chakra-progress__indicator css-qa0lra"
              cx="50"
              cy="50"
              r="42"
              stroke-width="10px"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Data Table Page Should sort by name 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-f7fu58"
      >
        <div
          class="css-8zasf"
        >
          <div
            class="css-1wo2j8q"
          >
            <div
              class="css-1anpw3e"
            >
              <svg
                class="chakra-icon css-tud5pv"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                  fill="currentColor"
                />
              </svg>
              <input
                class="chakra-input css-oce1e7"
                id="search:r2:"
                placeholder="Search..."
                value=""
              />
            </div>
          </div>
          <button
            class="chakra-button css-11ko6ne"
            type="button"
          >
            + Add customer
          </button>
        </div>
        <div
          class="css-1milhtq"
        >
          <div
            class="css-1dj3a23"
          >
            <p
              class="chakra-text css-0"
            >
              Name
            </p>
            <img
              class="chakra-image css-0"
              src="/icons/column-sorting-none.svg"
            />
          </div>
          <p
            class="chakra-text css-1i8mt9"
          >
            Description
          </p>
          <p
            class="chakra-text css-l7sbe5"
          >
            Status
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Rate
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Balance
          </p>
          <p
            class="chakra-text css-mxka6n"
          >
            Deposit
          </p>
          <div
            class="css-yikrym"
          />
        </div>
        <div
          class="css-3ehqor"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            class="chakra-progress css-7d3cb9"
            data-indeterminate=""
            role="progressbar"
          >
            <svg
              class="css-tn9112"
              viewBox="0 0 100 100"
            >
              <circle
                class="chakra-progress__track css-1bhxzgy"
                cx="50"
                cy="50"
                r="42"
                stroke-width="10px"
              />
              <circle
                class="chakra-progress__indicator css-qa0lra"
                cx="50"
                cy="50"
                r="42"
                stroke-width="10px"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-f7fu58"
    >
      <div
        class="css-8zasf"
      >
        <div
          class="css-1wo2j8q"
        >
          <div
            class="css-1anpw3e"
          >
            <svg
              class="chakra-icon css-tud5pv"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
                fill="currentColor"
              />
            </svg>
            <input
              class="chakra-input css-oce1e7"
              id="search:r2:"
              placeholder="Search..."
              value=""
            />
          </div>
        </div>
        <button
          class="chakra-button css-11ko6ne"
          type="button"
        >
          + Add customer
        </button>
      </div>
      <div
        class="css-1milhtq"
      >
        <div
          class="css-1dj3a23"
        >
          <p
            class="chakra-text css-0"
          >
            Name
          </p>
          <img
            class="chakra-image css-0"
            src="/icons/column-sorting-none.svg"
          />
        </div>
        <p
          class="chakra-text css-1i8mt9"
        >
          Description
        </p>
        <p
          class="chakra-text css-l7sbe5"
        >
          Status
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Rate
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Balance
        </p>
        <p
          class="chakra-text css-mxka6n"
        >
          Deposit
        </p>
        <div
          class="css-yikrym"
        />
      </div>
      <div
        class="css-3ehqor"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="chakra-progress css-7d3cb9"
          data-indeterminate=""
          role="progressbar"
        >
          <svg
            class="css-tn9112"
            viewBox="0 0 100 100"
          >
            <circle
              class="chakra-progress__track css-1bhxzgy"
              cx="50"
              cy="50"
              r="42"
              stroke-width="10px"
            />
            <circle
              class="chakra-progress__indicator css-qa0lra"
              cx="50"
              cy="50"
              r="42"
              stroke-width="10px"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
